### PR TITLE
Add different icons for traffic indicators

### PR DESCRIPTION
**Pull request by kibidev**
Based on MAVLink ADSB_EMITTER_TYPE, currently support different icons for
plane, helicopter, drone, parachute/paraglider and unknown.

Screenshot from 2024-03-19 13-43-57

### CODE REVIEW

File: src/ADSB/ADSBVehicle.cc [hunk 1]
```diff
@@ -30,6 +30,14 @@ void ADSBVehicle::update(const VehicleInfo_t& vehicleInfo)
         qCWarning(ADSBVehicleManagerLog) << "ICAO address mismatch expected:actual" << _icaoAddress << vehicleInfo.icaoAddress;
         return;
     }
+
+    if(vehicleInfo.emitter != _emitter) {
+	if (vehicleInfo.emitter != _emitter) {
+            _emitter = vehicleInfo.emitter;
+            emit emitterChanged();
+        }
+    }
```
Comment (hermankolden) [inline src/ADSB/ADSBVehicle.cc:39]: Redundant if-statement

File: src/ADSB/ADSBVehicle.h [hunk 2]
```diff
@@ -20,6 +20,31 @@ class ADSBVehicle : public QObject
     Q_OBJECT
 
 public:
+    enum class Emitter : uint8_t {
```
Comment (hermankolden) [inline src/ADSB/ADSBVehicle.h:23]: Emitter wasn't clear to me what it meant when I saw it elsewhere. Can we call it EmitterType instead?

File: src/FlightMap/MapItems/VehicleMapItem.qml [hunk 3]
```diff
@@ -16,16 +16,18 @@ import QGroundControl               1.0
...
-    property double heading:        vehicle ? vehicle.heading.value : Number.NaN    ///< Vehicle heading, NAN for none
-    property real   size:           _adsbVehicle ? _adsbSize : _uavSize             /// Size for icon
-    property bool   alert:          false                                           /// Collision alert
+    property var     vehicle                                                         /// Vehicle object, undefined for ADSB vehicle
+    property var     map
+    property double  altitude:       Number.NaN                                      ///< NAN to not show
+    property string  callsign:       ""                                              ///< Vehicle callsign
+    property double  heading:        vehicle ? vehicle.heading.value : Number.NaN    ///< Vehicle heading, NAN for none
+    property real    size:           _adsbVehicle ? _adsbSize : _uavSize             /// Size for icon
+    property bool    alert:          false                                           /// Collision alert
```
Comment (hermankolden) [inline src/FlightMap/MapItems/VehicleMapItem.qml:29]: Unclear to me what the change was here, seems like only whitespace

File: src/QGCApplication.cc [hunk 4]
```diff
@@ -103,6 +103,7 @@
 #include "QGCMAVLink.h"
 #include "VehicleLinkManager.h"
 #include "Autotune.h"
+#include "ADSBVehicle.h"
```
Comment (hermankolden) [inline src/QGCApplication.cc:106]: Unused I think

**Review by hermankolden — CHANGES_REQUESTED** [review]
Looks good! Some comments for cleanup

File: src/ADSB/ADSBVehicle.cc (hunk 1 above)
Comment (kibidev) [reply src/ADSB/ADSBVehicle.cc]: Fixed

File: src/ADSB/ADSBVehicle.h (hunk 2 above)
Comment (kibidev) [reply src/ADSB/ADSBVehicle.h]: Changed

File: src/FlightMap/MapItems/VehicleMapItem.qml (hunk 3 above)
Comment (kibidev) [reply src/FlightMap/MapItems/VehicleMapItem.qml]: Yes, it was for alignment changes that is no longer needed. Removed now.

File: src/QGCApplication.cc (hunk 4 above)
Comment (kibidev) [reply src/QGCApplication.cc]: yes, removed

File: src/ADSB/ADSBVehicleManager.cc [hunk 5]
```diff
@@ -18,6 +18,7 @@
 ADSBVehicleManager::ADSBVehicleManager(QGCApplication* app, QGCToolbox* toolbox)
     : QGCTool(app, toolbox)
 {
+    qmlRegisterUncreatableType<ADSBVehicle>("QGroundControl.ADSBVehicle",1, 0, "ADSBVehicle","Reference only");
```
Comment (hermankolden) [inline src/ADSB/ADSBVehicleManager.cc:21]: I didn't notice this until now, I have no idea what it is. I think it came from Markus' commit

**Review by hermankolden — APPROVED** [review]
Great! I have a comment that doesn't need to be handled if you think it's OK

**Pull request merged**